# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Tue Apr 07 15:17:14 2020 +0000
+++ b/dom/base/nsFrameLoader.cpp	Tue Apr 07 15:17:47 2020 +0000
@@ -2472,27 +2472,24 @@
   return lazyHeight;
 }
 
-static Tuple<ContentParent*, BrowserParent*> GetContentParent(
-    Element* aBrowser) {
-  using ReturnTuple = Tuple<ContentParent*, BrowserParent*>;
-
+static ContentParent* GetContentParent(Element* aBrowser) {
   nsCOMPtr<nsIBrowser> browser = aBrowser ? aBrowser->AsBrowser() : nullptr;
   if (!browser) {
-    return ReturnTuple(nullptr, nullptr);
+    return nullptr;
   }
 
   RefPtr<nsFrameLoader> otherLoader;
   browser->GetSameProcessAsFrameLoader(getter_AddRefs(otherLoader));
   if (!otherLoader) {
-    return ReturnTuple(nullptr, nullptr);
+    return nullptr;
   }
 
   BrowserParent* browserParent = BrowserParent::GetFrom(otherLoader);
-  if (browserParent && browserParent->Manager()) {
-    return MakeTuple(browserParent->Manager(), browserParent);
+  if (browserParent) {
+    return browserParent->Manager();
   }
 
-  return ReturnTuple(nullptr, nullptr);
+  return nullptr;
 }
 
 bool nsFrameLoader::EnsureRemoteBrowser() {
@@ -2597,21 +2594,7 @@
     }
 
     // Try to get the related content parent from our browser element.
-    Tie(openerContentParent, sameTabGroupAs) = GetContentParent(mOwnerContent);
-    // If we have an opener, it may be in the same process as our new child, and
-    // therefore needs to be in the same tab group. The long term solution to
-    // this problem is to get rid of TabGroups entirely. In the short term, the
-    // following hack deals with the specific problem that when a window has an
-    // opener, it asserts that its BrowserChild is bound to the same tab group
-    // as its opener. There are likely other mismatches that it does not handle,
-    // but those will all be fixed by the removal of TabGroups.
-    if (RefPtr<BrowsingContext> openerBC =
-            mPendingBrowsingContext->GetOpener()) {
-      auto global = openerBC->Canonical()->GetCurrentWindowGlobal();
-      if (global) {
-        sameTabGroupAs = global->GetBrowserParent();
-      }
-    }
+    openerContentParent = GetContentParent(mOwnerContent);
   }
 
   uint32_t chromeFlags = 0;
@@ -2656,7 +2639,7 @@
 
   mRemoteBrowser = ContentParent::CreateBrowser(
       context, ownerElement, mRemoteType, mPendingBrowsingContext,
-      openerContentParent, sameTabGroupAs, nextRemoteTabId);
+      openerContentParent, nextRemoteTabId);
   if (!mRemoteBrowser) {
     return false;
   }